### PR TITLE
doc update + mbed-os 5 example application

### PR DESCRIPTION
**Pull request by jupe**
## Status
**READY**

## Description

Update documentation a bit and add mbed-os 5 demonstrate application

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,14 +20,14 @@ The purpose of the library is to provide a light, simple and general tracing sol
 ## Compromises
 
 * The traces are stored as ASCII arrays in the flash memory (pretty high memory consumption). Therefore, it is not necessary to:
- * encode/decode the trace messages on the fly (this may take too much CPU time) or 
- * have external dev-env dependencies to encode the traces compile time and an external application to decode the traces. 
+  * encode/decode the trace messages on the fly (this may take too much CPU time) or
+  * have external dev-env dependencies to encode the traces compile time and an external application to decode the traces.
```
Comment (tommikas) [inline README.md]: Were the above extra spaces intentional?

File: README.md [hunk 2]
```diff
@@ -20,14 +20,14 @@ The purpose of the library is to provide a light, simple and general tracing sol
...
-* The trace function uses `stdout` as the default output target because it goes directly to serial port when initialized. 
+* The trace function uses `stdout` as the default output target because it goes directly to serial port in mbed-os.
 * The trace function produces traces like: `[<levl>][grp ]: msg`. This provides an easy way to detect trace prints and separate traces from normal prints (for example with _regex_).
 * This approach requires a `sprintf` implementation (`stdio.h`). The memory consumption is pretty high, but it allows an efficient way to format traces.
 * The solution is not Interrupt safe. (PRs are more than welcome.)
-* The solution is not Thread safe by default. Thread safety for the actual trace calls can be enabled by providing wait and release callback functions that use mutexes defined by the application. 
+* The solution is not Thread safe by default. Thread safety for the actual trace calls can be enabled by providing wait and release callback functions that use mutexes defined by the application.
```
Comment (tommikas) [inline README.md]: Removing the extra whitespace is ok, but I wonder if the T in the first "Thread" should also be made a small one... "Thread" in the sense of an OS thread shouldn't be capitalized, and "Thread" with a capital t is the name of a network protocol (among other things). Of course the context clues should eventually clarify things either way, but still.

File: example/mbed-os-5/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+```
+mbed deploy
+mbed compile -t GCC_ARM -m K64F
+```
+
+
+## Usage
+
+When you flash target with this application and open terminal you should see following traces:
```
Comment (tommikas) [inline example/mbed-os-5/README.md]: "**a** target", "**a** terminal"  and "**the** following traces"

File: example/mbed-os-5/main.c [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ * limitations under the License.
+ */
+#include <stdio.h>
+
+#include "mbed.h"
+
+#define TRACE_GROUP                 "main"
+#define DEFAULT_SERIAL_BAUD_RATE    (115200)
+
+static RawSerial pc(USBTX, USBRX, DEFAULT_SERIAL_BAUD_RATE);
```
Comment (tommikas) [inline example/mbed-os-5/main.c]: I don't think RawSerial should be used. Better to just define the baud rate in the app config file.

File: example/mbed-os-5/main.c [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <stdio.h>
+
+#include "mbed.h"
```
Comment (tommikas) [inline example/mbed-os-5/main.c]: Missing `mbed_trace.h` include

File: example/mbed-os-5/mbed_app.json [hunk 6]
```diff
@@ -0,0 +1,12 @@
+{
+    "macros": [
+        "MEM_ALLOC=malloc",
+        "MEM_FREE=free"
+    ],
+    "target_overrides": {
+        "*": {
+            "target.features_add": ["COMMON_PAL"],
```
Comment (tommikas) [inline example/mbed-os-5/mbed_app.json]: `"platform.stdio-baud-rate": 115200,` here should be enough.

`"platform.stdio-convert-newlines": true,`, and possibly even `"platform.stdio-buffered-serial": true,` might be worth configuring too, unless the idea is to have the absolute bare minimum.

File: README.md (hunk 1 above)
Comment (jupe) [reply README.md]: yes - intended because related to previous topic..

**Pull request merged**